pkg/network/server_test.go

```diff
@@ -4,29 +4,33 @@ import (
 	"net"
 	"testing"
 
+	"github.com/nspcc-dev/neo-go/pkg/network/capability"
 	"github.com/nspcc-dev/neo-go/pkg/network/payload"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func TestSendVersion(t *testing.T) {
 	var (
-		s = newTestServer(t)
+		s = newTestServer(t, ServerConfig{Port: 0, UserAgent: "/test/"})
 		p = newLocalPeer(t, s)
 	)
-	s.Port = 3000
-	s.UserAgent = "/test/"
-
 	p.messageHandler = func(t *testing.T, msg *Message) {
 		assert.Equal(t, CMDVersion, msg.Command)
 		assert.IsType(t, msg.Payload, &payload.Version{})
 		version := msg.Payload.(*payload.Version)
 		assert.NotZero(t, version.Nonce)
-		assert.Equal(t, uint16(3000), version.Port)
-		assert.Equal(t, uint64(1), version.Services)
+		assert.Equal(t, 1, len(version.Capabilities))
+		assert.ElementsMatch(t, []capability.Capability{
+			{
+				Type: capability.TCPServer,
+				Data: &capability.Server{
+					Port: 0,
```

Comment:
Is it really zero that is being sent from the server?

Reply:
That's the case when we don't have `listener` set in our `Transporter`, that's why we've got the port from server configuration, which is set to be 0 (13 lines above). Is it desired behaviour?

Reply:
I'd expect a server configured with port `0` to send its actual port anyway if it sets up proper TCP listener.